samples: matter: Fix the Last Fabric Removal feature (#12698)
nrfconnect/sdk-nrf: doc/nrf/protocols/matter/getting_started/advanced_kconfigs.rst

* :kconfig:option:`CONFIG_CHIP_LAST_FABRIC_REMOVED_ERASE_AND_PAIRING_START` - Remove all saved network credentials and start Bluetooth LE advertising.
  The device will remove all saved network credentials, keep application-specific non-volatile data, and start advertising Bluetooth LE Matter service.
  After that, it will be ready for commissioning to Matter over Bluetooth LE.

>   This option is not available in the :ref:`matter_template_sample` sample.

Reviewer: Actually why is that?

Author: After the f2f discussion, I've decided to add it to Matter Template to fix the build there.

>   This option is not available on :ref:`Matter Template sample <matter_template_sample>`

Reviewer 2: ```suggestion
  This option is not available in the :ref:`matter_template_sample` sample.
```

* :kconfig:option:`CONFIG_CHIP_LAST_FABRIC_REMOVED_ERASE_AND_REBOOT` - Remove all saved network credentials and reboot the device.
  This option is selected by default.

  When the :kconfig:option:`CONFIG_CHIP_FACTORY_RESET_ERASE_NVS` Kconfig option is also set to ``y``, the device will also remove all non-volatile data stored on the device, including application-specific entries.
  This means the device is restored to the factory settings.

> To create a delay between  the chosen reaction and the last fabric being removed, set the :kconfig:option:`CONFIG_CHIP_LAST_FABRIC_REMOVED_SCHEDULE_TIME` Kconfig option to a specific time in milliseconds.

Reviewer: Btw. I noticed the links to these `CONFIG_CHIP_LAST_FABRIC_REMOVED_...` configs does not work. Could you take a look on that? For some reason links to all other configs on this page work properly.

Reviewer: Ok, I know why it works like this, and the problem is bigger than I thought and not related strictly to this PR. I will create a separate task to track it.

Author: Right, it seems this doesn't work when we have Kconfigs defined in the Kconfig.features file...

> To make a delay before doing the chosen action after removing the last fabric set the :kconfig:option:`CONFIG_CHIP_LAST_FABRIC_REMOVED_SCHEDULE_TIME` Kconfig option to a specific time in milliseconds.

Reviewer 2: ```suggestion
To create a delay between  the chosen reaction and the last fabric being removed, set the :kconfig:option:`CONFIG_CHIP_LAST_FABRIC_REMOVED_SCHEDULE_TIME` Kconfig option to a specific time in milliseconds.
```

> By default this Kconfig option is set to 500 milliseconds.

Reviewer: Minor: Isn't linking to kconfig option enough? I mean, the user can click on the link and see the default value. And when the value is directly given in the text, we will have to update docs after modifying the Kconfig value.

Author: Since we have a problem with linking, I think we should leave it as it is.  Shouldn't we?

Reviewer: If the fix for links will not come shortly I'm ok with leaving it as it is.

  When the :kconfig:option:`CONFIG_CHIP_FACTORY_RESET_ERASE_NVS` Kconfig option is also set to ``y``, the device will also remove all non-volatile data stored on the device, including application-specific entries.
  This means the device is restored to the factory settings.

> To create a delay between  the chosen reaction and the last fabric being removed, set the :kconfig:option:`CONFIG_CHIP_LAST_FABRIC_REMOVED_SCHEDULE_TIME` Kconfig option to a specific time in milliseconds.

Reviewer: You didn't modify the Kconfig name accordingly to the recent changes in corresponding PR.

Author: Fixed.
